stats/stats.go

```diff
@@ -0,0 +1,140 @@
+package stats
+
+import (
+	"context"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/elliotchance/orderedmap"
+	"github.com/hashicorp/go-hclog"
+	"github.com/segmentio/stats/v4"
+)
+
+type Measure struct {
+	id       string
+	duration time.Duration
+	stamp    bool
+	time     time.Time
+}
+
+type Stat struct {
+	Start    time.Time
+	Duration time.Duration
+}
+
+type LogHandler struct {
+	logger   hclog.Logger
+	measures chan Measure
+	stats    *orderedmap.OrderedMap
+	mu       sync.Mutex
+}
+
+type Options struct {
+	Logger  hclog.Logger
+	Tick    time.Duration
+	Handler stats.Handler
+}
```

Comment:
Ah, thanks I followed the existing pattern in the code from https://github.com/cloudquery/cq-provider-sdk/blob/be60d7430a62f4b1d328c05b193ce55dd01c6fd1/serve/debug.go#L35.

Let me use the one you sent

Reply:
Update:

@roneli can you clarify how the function options pattern fit in when there is not constructor?
I can add one, but it might be confusing since this package is meant to be used without it.

Reply:
Ah, I think I understand, thanks for clarifying.
Is this https://github.com/cloudquery/cq-provider-sdk/pull/237/commits/67b759131cecb602d6a2226f3b59b776262f71d9 what you had in mind?

I kept the context as logger as arguments as those are required